### ISSUE TITLE
Unexpected Result with --optimize-no-op

### ISSUE THREAD

**Issue by Binary-Bear** [TheWorkingDeveloper/ruleprocessorY#12]
Using Win7 64
ruleprocessorY.exe 1.3
General_Rules.rule [Attached](https://github.com/0xVavaldi/ruleprocessorY/files/15082068/General_Rules.zip)

Command used
`ruleprocessorY.exe --optimize-no-op --hashcat-input --hashcat-output --optimize-debug Debug.txt -r General_Rules.rule>>General_Rules_optimized.rule`

I placed a group of general rules into a file. (General_Rules.rule)

I ran ruleprocessorY.exe with the "optimize-no-op" option.

I also chose to output to a debug file.

When ruleprocessorY.exe started the following was displayed in the command screen (not the debug file).

```
Unknown rule function [94413]: "0"
Unknown rule function [94414]: "0"
Unknown rule function [233095]: "0"
Unknown rule function [271297]: "0"
Unknown rule function [274805]: "0"
Unknown rule function [325264]: "0"
Unknown rule function [325267]: "0"
Unknown rule function [325268]: "0"
```

The debug.txt file was empty.

I deleted the original "General_Rules.rule" and I ran ruleprocessorY.exe on the newly created "General_Rules_optimized.rule" expecting there to be no issues found.

However when ruleprocessorY.exe started the following was displayed in the command screen (not the debug file).

```
Invalid Rule [2879]: "s\x20\    /"
Unknown rule function [4223]: "\"
Unknown rule function [39763]: "\"
Unknown rule function [44280]: "\"
Unknown rule function [90375]: "\"
Unknown rule function [90376]: "\"
Invalid Rule [109822]: "*05     s\x20\  /"
Invalid Rule [110796]: "*13     s\x20\  /"
Invalid Rule [114155]: "*40     s\x20\  /"
Unknown rule function [122559]: "\"
Unknown rule function [146455]: "\"
Unknown rule function [146456]: "\"
Unknown rule function [146457]: "\"
Invalid Rule [185718]: "[       ^C      s\x20\  /"
Invalid Rule [214004]: "K       xB9     s\x20\  /"
Unknown rule function [248536]: "\"
Unknown rule function [263715]: "\"
Unknown rule function [269969]: "\"
Unknown rule function [272078]: "\"
Invalid Rule [273935]: "sF>     Y3      s\x20\  /"
Unknown rule function [298487]: "\"
Unknown rule function [312715]: "\"
Unknown rule function [312716]: "\"
Invalid Rule [315266]: "sS@     sO.     s\x20\  /"
```

The debug.txt file was empty.

The expected output following the second run with ruleprocessorY.exe would be no unknown or invalid rules displayed within the command screen.

It appears that ruleprocessorY.exe might be creating invalid rules during the "optimize-no-op" process.

**Comment by 0xVavaldi** [comment]
Could you find out what the original rules are on line 2879, 109822, and 110796 for example? Relating to your original General_Rules.rule file.

I am also curious what 4223 and 39763 are
